Dockerfile

```diff
@@ -1,25 +1,26 @@
 FROM ubuntu:16.04
 MAINTAINER Matthew Bruzek <Matthew.Bruzek@canonical.com>
 
-RUN useradd -m ubuntu -s /bin/bash
+ARG JUJU_USER=ubuntu
 
-RUN mkdir -p /home/ubuntu/.local/share/juju 
-RUN mkdir -p /home/ubuntu/charms
+ENV JUJU_DATA /home/$JUJU_USER/.local/share/juju
+ENV JUJU_REPOSITORY /home/$JUJU_USER/charms
 
-RUN chown -R ubuntu:ubuntu /home/ubuntu/.local
-RUN chown -R ubuntu:ubuntu /home/ubuntu/charms
+RUN useradd -m $JUJU_USER -s /bin/bash
 
-VOLUME [ "/home/ubuntu/.local/share/juju", "/home/ubuntu/charms" ]
+RUN mkdir -p $JUJU_DATA
+RUN mkdir -p $JUJU_REPOSITORY
+
+RUN chown -R $JUJU_USER:$JUJU_USER /home/$JUJU_USER/.local
+RUN chown -R $JUJU_USER:$JUJU_USER $JUJU_REPOSITORY
+
+VOLUME [ "$JUJU_DATA", "$JUJU_REPOSITORY" ]
 
 ADD setup.sh /setup.sh
```

Comment:
as with #20, you're `ADD` instead of `COPY`.  i don't mind it so much, just curious if this was an oversight.  whatever you go with, i'll be happy if `master` and `juju-1` are consistent.